openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -368,3 +375,93 @@ def test_xblock_handlers(self, xblock_type, xblock_name, user, status_code):
         response = self.client.post(url)
         self.assertEqual(response.status_code, status_code)
+
+
+@ddt.ddt
+@override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+@override_settings(FIELD_OVERRIDE_PROVIDERS=(
+    'openedx.features.content_type_gating.field_override.ContentTypeGatingFieldOverride',
+))
+class TestConditionalContentAccess(TestConditionalContent):
+    @classmethod
+    def setUpClass(cls):
+        super(TestConditionalContentAccess, cls).setUpClass()
+        cls.factory = RequestFactory()
+    
+    def setUp(self):
+        super(TestConditionalContentAccess, self).setUp()
+
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='audit')
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='verified')
+
+        # Audit users have already been created. Now create verified users, and partition.
+        self.student_verified_a = UserFactory.create(username='student_verified_a', email='student_verified_a@example.com')
+        CourseEnrollmentFactory.create(user=self.student_verified_a, course_id=self.course.id, mode='verified')
+        self.student_verified_b = UserFactory.create(username='student_verified_b', email='student_verified_b@example.com')
+        CourseEnrollmentFactory.create(user=self.student_verified_b, course_id=self.course.id, mode='verified')
+
+        # Use this to put users into content gating groups
+        UserCourseTagFactory(
+            user=self.student_verified_a,
+            course_id=self.course.id,
+            key='xblock.partition_service.partition_{0}'.format(self.partition.id),
+            value=str(self.user_partition_group_a)
+        )
+        UserCourseTagFactory(
+            user=self.student_verified_b,
+            course_id=self.course.id,
+            key='xblock.partition_service.partition_{0}'.format(self.partition.id),
+            value=str(self.user_partition_group_b)
+        )
+
+        self.block_a = ItemFactory.create(
+            parent=self.vertical_a,
+            category='problem',
+            display_name='problem_a',
+            graded=True,
+            weight=1,
+        )
+        self.block_b = ItemFactory.create(
+            parent=self.vertical_b,
+            category='problem',
+            display_name='problem_b',
+            graded=True,
+            weight=1,
+        )
+
+
+    @ddt.unpack
+    def test_access_based_on_conditional_content(self):
+        """
+         If a user is enrolled as an audit user they should not have access to graded problems, including conditional content.  
+         All paid type tracks should have access graded problems including conditional content.
+         """
+
+        _assert_block_is_gated(
+            block=self.block_a,
+            user_id=self.student_a.id,
+            course_id=self.course.id,
+            is_gated=True,
+            request_factory=self.factory,
+        )
+        _assert_block_is_gated(
```

Comment:
I had looked this over and suggested leaving this as is -- I'm not convinced the ddt solution is actually more readable.

Reply:
I could make a dictionary inside of the function, and loop through it, but I feel like that would not be as clean.

Reply:
since Gabe suggested leaving it as is, I recommend leaving as is.  
the comment was more of a nitpick/personal preference, but this solution will be perfectly clear if you add a one-line comment above each test case explaining which one it is

if you're curious about general engineering feedback on tests, here is my high level opinion about cases like this (which may not be shared by other engineers):
Using ddt/for loops in a test is not typically clearer because the actual code is simpler, even if it is more DRY (in fact sometimes this makes the code more complicated).   
However, I do think it makes tests clearer because it forces the developer to pull out all the parameters that differentiate each test case. 
When someone with no familiarity with the code looks at a test using ddt for example, they can clearly tell what the test cases being tested are based on the list of parameter lists. When you have one block of code you actually need to grok all the code to understand the test cases.  
I've seen many tests where over time the test becomes one giant block of code and it can be difficult to trace back through each variable to figure out which ones are shared among different assert statements and which ones differ